src/CustomType/Data/Slice/Slice.php

```diff
@@ -42,8 +42,8 @@ final public function toArray () : array
 			"fieldset" => $this->label,
 			"description" => $this->description,
 			"icon" => $this->icon,
-			"non-repeat" => KeyedMapHelper::transformKeyedListOfTypes($this->fields),
-			"repeat" => KeyedMapHelper::transformKeyedListOfTypes($this->repeatedFields),
+			"non-repeat" => !empty($this->fields) ? KeyedMapHelper::transformKeyedListOfTypes($this->fields) : null,
+			"repeat" => !empty($this->repeatedFields) ? KeyedMapHelper::transformKeyedListOfTypes($this->repeatedFields) : null,
```

Comment:
Instead you can just adapt your `filterOptionalFields()` to filter out empty arrays?